.pre-commit-config.yaml

```diff
@@ -0,0 +1,40 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+#
+# Install pre-commit itself:
+#   $ pip install pre-commit
+#     OR
+#   $ brew install pre-commit
+#
+# Update all pre-commit hooks to latest version:
+#   $ pre-commit autoupdate
+#
+# Install all pre-commit hooks locally:
+#   $ pre-commit install
```

Comment:
Shall we actually perform that automatically with the test script so that the pre-commit hook is triggered locally as well? It would be great to see the failures / errors before committing changes and not have to wait for the results in CI.

Reply:
Yes, this is feasible.

However I don't know if it's worth the effort, it depends on jgraham@, I guess :-)

Reply:
Yes, looks like he doesn't want it by default, but maybe if we keep at least some of the checks we could have an argument for the scripts to opt-in for those checks.